lib/wits/helpers.rb

```diff
@@ -42,14 +42,26 @@ def format_prices(node, date, prices, price_type)
       }
     end
 
-    def parse_nz_time(date, time)
-      tz = TZInfo::Timezone.get('Pacific/Auckland')
+    def parse_nz_time(date, time, repeated_time)
+      date = Date.parse(date) unless date.is_a?(Date)
 
-      # Timezone information of the Time object is ignored by TZInfo
+      # TimezonePeriod for local Time (midnight)
+      tz_period = nz_time_zone.period_for_local(date.to_time)
+
+      # Assume we transitioned between NZST and NZDT if
+      # a time is repeated
+      dst = tz_period.dst?
+      dst = !dst if repeated_time
+
+      # Time zone information of the Time object is ignored by TZInfo
       time = Time.parse("#{date} #{time}")
 
       # Output in local time
-      tz.local_to_utc(time).localtime
+      nz_time_zone.local_to_utc(time, dst).localtime
+    end
+
+    def nz_time_zone
+      @nz_time_zone ||= TZInfo::Timezone.get('Pacific/Auckland')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.